{
  "cells": [
    {
      "cell_type": "code",
      "execution_count": 1,
      "id": "9fe1c62b",
      "metadata": {},
      "outputs": [],
      "source": [
        "import pandas as pd\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "id": "582885af",
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/plain": [
              "Index(['ID', 'Dataset', 'Gene', 'HGNC_id', 'Chrom', 'STRAND', 'hg19_pos',\n",
              "       'hg38_start', 'hg38_end', 'ref_allele',\n",
              "       ...\n",
              "       'revel_train_amino', 'mp2_train_amino',\n",
              "       'Ref_seq_transcript_ID_stripped', 'clinvar_conflict_flag',\n",
              "       'splice_var_amino', 'condensed_consequence', 'abundance_avg',\n",
              "       'gmm_density_abnormal', 'gmm_density_normal', 'functional_consequence'],\n",
              "      dtype='object', length=143)"
            ]
          },
          "execution_count": null,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "df = pd.read_csv('GMM_classifications/TSC2.lib1.gmm.tsv', sep='\\t')\n",
        "\n",
        "df.head()\n",
        "\n",
        "df.columns\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "id": "8a4eb75f",
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "['ID', 'Dataset', 'Gene', 'HGNC_id', 'Chrom', 'STRAND', 'hg19_pos', 'hg38_start', 'hg38_end', 'ref_allele', 'alt_allele', 'auth_transcript_id', 'transcript_pos', 'transcript_ref', 'transcript_alt', 'aa_pos', 'aa_ref', 'aa_alt', 'hgvs_c', 'hgvs_p', 'consequence', 'simplified_consequence', 'auth_reported_score', 'auth_reported_rep_score', 'auth_reported_func_class', 'splice_measure', 'gnomad_MAF', 'clinvar_sig', 'clinvar_star', 'clinvar_date_last_reviewed', 'nucleotide_or_aa', 'MaveDB Score Set URN', 'Ensembl_transcript_ID', 'Ref_seq_transcript_ID', 'Model_system', 'Assay Type', 'Phenotype Measured ontology term', 'Molecular or Biological Process Investigated (GO term)', 'IGVF_produced', 'Interval 1 name', 'Interval 1 range', 'Interval 1 MaveDB class', 'Interval 2 name', 'Interval 2 range', 'Interval 2 MaveDB class', 'Interval 3 name', 'Interval 3 range', 'Interval 3 MaveDB class', 'Interval 4 name', 'Interval 4 range', 'Interval 4 MaveDB class', 'Interval 5 name', 'Interval 5 range', 'Interval 5 MaveDB class', 'Interval 6 name', 'Interval 6 range', 'Interval 6 MaveDB class', 'Flag', 'REVEL', 'AM_score', 'AM_class', 'spliceAI_DS_AG', 'spliceAI_DS_AL', 'spliceAI_DS_DG', 'spliceAI_DS_DL', 'spliceAI_DP_AG', 'spliceAI_DP_AL', 'spliceAI_DP_DG', 'spliceAI_DP_DL', 'REVEL_train', 'MutPred2', 'MP2_train', 'ClinVar Variation Id_ClinGen_repo', 'Allele Registry Id_ClinGen_repo', 'Disease_ClinGen_repo', 'Mondo Id_ClinGen_repo', 'Mode of Inheritance_ClinGen_repo', 'Assertion_ClinGen_repo', 'Applied Evidence Codes (Met)_ClinGen_repo', 'Applied Evidence Codes (Not Met)_ClinGen_repo', 'Summary of interpretation_ClinGen_repo', 'PubMed Articles_ClinGen_repo', 'Expert Panel_ClinGen_repo', 'Guideline_ClinGen_repo', 'Approval Date_ClinGen_repo', 'Published Date_ClinGen_repo', 'Retracted_ClinGen_repo', 'Evidence Repo Link_ClinGen_repo', 'Uuid_ClinGen_repo', 'Updated_Classification_ClinGen_repo', 'Updated_Evidence Codes_ClinGen_repo', 'scoreset_name', 'inverted', 'pathogenic_threshold_1', 'pathogenic_threshold_2', 'pathogenic_threshold_3', 'pathogenic_threshold_4', 'pathogenic_threshold_5', 'benign_threshold_1', 'benign_threshold_2', 'benign_threshold_3', 'benign_threshold_4', 'benign_threshold_5', 'dan_points', 'PP3_REVEL', 'BP4_REVEL', 'PP3_Mutpred2', 'BP4_Mutpred2', 'PP3_AM', 'BP4_AM', 'func_points', 'p_points_REVEL', 'total_points_REVEL', 'p_points_Mutpred2', 'total_points_Mutpred2', 'p_points_AM', 'total_points_AM', 'points_class_REVEL', 'points_class_Mutpred2', 'points_class_AM', 'split_zero_REVEL', 'split_zero_Mutpred2', 'split_zero_AM', 'split_zero_class_REVEL', 'split_zero_class_Mutpred2', 'split_zero_class_AM', 'splice_variant', 'Fxn_use_variant', 'REVEL_max', 'AM_max', 'Mutpred2_max', 'clnsig_group', 'gnomad_seen', 'revel_train_amino', 'mp2_train_amino', 'Ref_seq_transcript_ID_stripped', 'clinvar_conflict_flag', 'splice_var_amino', 'condensed_consequence', 'abundance_avg', 'gmm_density_abnormal', 'gmm_density_normal', 'functional_consequence']\n"
          ]
        }
      ],
      "source": [
        "print(df.columns.tolist())"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "id": "1aa13071",
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>variant</th>\n",
              "      <th>abundance_1</th>\n",
              "      <th>abundance_2</th>\n",
              "      <th>abundance_avg</th>\n",
              "      <th>variance</th>\n",
              "      <th>std_dev</th>\n",
              "      <th>QC_flag</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>p.Val709Ter</td>\n",
              "      <td>-0.097886</td>\n",
              "      <td>-0.055560</td>\n",
              "      <td>-0.076723</td>\n",
              "      <td>0.000448</td>\n",
              "      <td>0.021163</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>p.Pro685Ter</td>\n",
              "      <td>-0.041725</td>\n",
              "      <td>0.047948</td>\n",
              "      <td>0.003111</td>\n",
              "      <td>0.002010</td>\n",
              "      <td>0.044836</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>p.Pro665Ter</td>\n",
              "      <td>0.097905</td>\n",
              "      <td>0.123814</td>\n",
              "      <td>0.110860</td>\n",
              "      <td>0.000168</td>\n",
              "      <td>0.012955</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>p.Ala583Ter</td>\n",
              "      <td>-0.016799</td>\n",
              "      <td>-0.074520</td>\n",
              "      <td>-0.045660</td>\n",
              "      <td>0.000833</td>\n",
              "      <td>0.028860</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>p.Glu755Ter</td>\n",
              "      <td>-0.037540</td>\n",
              "      <td>-0.077817</td>\n",
              "      <td>-0.057678</td>\n",
              "      <td>0.000406</td>\n",
              "      <td>0.020138</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3946</th>\n",
              "      <td>p.Glu715Trp</td>\n",
              "      <td>0.707929</td>\n",
              "      <td>0.923987</td>\n",
              "      <td>0.815958</td>\n",
              "      <td>0.011670</td>\n",
              "      <td>0.108029</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3947</th>\n",
              "      <td>p.Leu722Gln</td>\n",
              "      <td>0.795435</td>\n",
              "      <td>0.831680</td>\n",
              "      <td>0.813558</td>\n",
              "      <td>0.000328</td>\n",
              "      <td>0.018123</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3948</th>\n",
              "      <td>p.Ser735Met</td>\n",
              "      <td>0.856474</td>\n",
              "      <td>0.891518</td>\n",
              "      <td>0.873996</td>\n",
              "      <td>0.000307</td>\n",
              "      <td>0.017522</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3949</th>\n",
              "      <td>p.Ser687Thr</td>\n",
              "      <td>0.633401</td>\n",
              "      <td>0.714158</td>\n",
              "      <td>0.673780</td>\n",
              "      <td>0.001630</td>\n",
              "      <td>0.040378</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3950</th>\n",
              "      <td>p.Val709Pro</td>\n",
              "      <td>0.013122</td>\n",
              "      <td>0.048279</td>\n",
              "      <td>0.030700</td>\n",
              "      <td>0.000309</td>\n",
              "      <td>0.017579</td>\n",
              "      <td>True</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>3951 rows × 7 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "          variant  abundance_1  abundance_2  abundance_avg  variance  \\\n",
              "0     p.Val709Ter    -0.097886    -0.055560      -0.076723  0.000448   \n",
              "1     p.Pro685Ter    -0.041725     0.047948       0.003111  0.002010   \n",
              "2     p.Pro665Ter     0.097905     0.123814       0.110860  0.000168   \n",
              "3     p.Ala583Ter    -0.016799    -0.074520      -0.045660  0.000833   \n",
              "4     p.Glu755Ter    -0.037540    -0.077817      -0.057678  0.000406   \n",
              "...           ...          ...          ...            ...       ...   \n",
              "3946  p.Glu715Trp     0.707929     0.923987       0.815958  0.011670   \n",
              "3947  p.Leu722Gln     0.795435     0.831680       0.813558  0.000328   \n",
              "3948  p.Ser735Met     0.856474     0.891518       0.873996  0.000307   \n",
              "3949  p.Ser687Thr     0.633401     0.714158       0.673780  0.001630   \n",
              "3950  p.Val709Pro     0.013122     0.048279       0.030700  0.000309   \n",
              "\n",
              "       std_dev  QC_flag  \n",
              "0     0.021163     True  \n",
              "1     0.044836     True  \n",
              "2     0.012955     True  \n",
              "3     0.028860     True  \n",
              "4     0.020138     True  \n",
              "...        ...      ...  \n",
              "3946  0.108029     True  \n",
              "3947  0.018123     True  \n",
              "3948  0.017522     True  \n",
              "3949  0.040378     True  \n",
              "3950  0.017579     True  \n",
              "\n",
              "[3951 rows x 7 columns]"
            ]
          },
          "execution_count": null,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "simple_score = pd.read_csv('20260217_TSC2_lib1_scores_QC.csv')\n",
        "simple_score[[\"ENSP\", \"variant\"]] = simple_score[\"variant\"].str.split(\":\", n=1, expand=True)\n",
        "simple_score = simple_score.drop(columns=[\"ENSP\"])\n",
        "simple_score"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "id": "d60aa11a",
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>variant</th>\n",
              "      <th>abundance_1</th>\n",
              "      <th>abundance_2</th>\n",
              "      <th>abundance_avg</th>\n",
              "      <th>variance</th>\n",
              "      <th>std_dev</th>\n",
              "      <th>QC_flag</th>\n",
              "      <th>functional_consequence</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>p.Val709Ter</td>\n",
              "      <td>-0.097886</td>\n",
              "      <td>-0.055560</td>\n",
              "      <td>-0.076723</td>\n",
              "      <td>0.000448</td>\n",
              "      <td>0.021163</td>\n",
              "      <td>True</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>p.Pro685Ter</td>\n",
              "      <td>-0.041725</td>\n",
              "      <td>0.047948</td>\n",
              "      <td>0.003111</td>\n",
              "      <td>0.002010</td>\n",
              "      <td>0.044836</td>\n",
              "      <td>True</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>p.Pro665Ter</td>\n",
              "      <td>0.097905</td>\n",
              "      <td>0.123814</td>\n",
              "      <td>0.110860</td>\n",
              "      <td>0.000168</td>\n",
              "      <td>0.012955</td>\n",
              "      <td>True</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>p.Ala583Ter</td>\n",
              "      <td>-0.016799</td>\n",
              "      <td>-0.074520</td>\n",
              "      <td>-0.045660</td>\n",
              "      <td>0.000833</td>\n",
              "      <td>0.028860</td>\n",
              "      <td>True</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>p.Glu755Ter</td>\n",
              "      <td>-0.037540</td>\n",
              "      <td>-0.077817</td>\n",
              "      <td>-0.057678</td>\n",
              "      <td>0.000406</td>\n",
              "      <td>0.020138</td>\n",
              "      <td>True</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3946</th>\n",
              "      <td>p.Glu715Trp</td>\n",
              "      <td>0.707929</td>\n",
              "      <td>0.923987</td>\n",
              "      <td>0.815958</td>\n",
              "      <td>0.011670</td>\n",
              "      <td>0.108029</td>\n",
              "      <td>True</td>\n",
              "      <td>functionally_normal</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3947</th>\n",
              "      <td>p.Leu722Gln</td>\n",
              "      <td>0.795435</td>\n",
              "      <td>0.831680</td>\n",
              "      <td>0.813558</td>\n",
              "      <td>0.000328</td>\n",
              "      <td>0.018123</td>\n",
              "      <td>True</td>\n",
              "      <td>functionally_normal</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3948</th>\n",
              "      <td>p.Ser735Met</td>\n",
              "      <td>0.856474</td>\n",
              "      <td>0.891518</td>\n",
              "      <td>0.873996</td>\n",
              "      <td>0.000307</td>\n",
              "      <td>0.017522</td>\n",
              "      <td>True</td>\n",
              "      <td>functionally_normal</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3949</th>\n",
              "      <td>p.Ser687Thr</td>\n",
              "      <td>0.633401</td>\n",
              "      <td>0.714158</td>\n",
              "      <td>0.673780</td>\n",
              "      <td>0.001630</td>\n",
              "      <td>0.040378</td>\n",
              "      <td>True</td>\n",
              "      <td>functionally_normal</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3950</th>\n",
              "      <td>p.Val709Pro</td>\n",
              "      <td>0.013122</td>\n",
              "      <td>0.048279</td>\n",
              "      <td>0.030700</td>\n",
              "      <td>0.000309</td>\n",
              "      <td>0.017579</td>\n",
              "      <td>True</td>\n",
              "      <td>functionally_abnormal</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>3951 rows × 8 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "          variant  abundance_1  abundance_2  abundance_avg  variance  \\\n",
              "0     p.Val709Ter    -0.097886    -0.055560      -0.076723  0.000448   \n",
              "1     p.Pro685Ter    -0.041725     0.047948       0.003111  0.002010   \n",
              "2     p.Pro665Ter     0.097905     0.123814       0.110860  0.000168   \n",
              "3     p.Ala583Ter    -0.016799    -0.074520      -0.045660  0.000833   \n",
              "4     p.Glu755Ter    -0.037540    -0.077817      -0.057678  0.000406   \n",
              "...           ...          ...          ...            ...       ...   \n",
              "3946  p.Glu715Trp     0.707929     0.923987       0.815958  0.011670   \n",
              "3947  p.Leu722Gln     0.795435     0.831680       0.813558  0.000328   \n",
              "3948  p.Ser735Met     0.856474     0.891518       0.873996  0.000307   \n",
              "3949  p.Ser687Thr     0.633401     0.714158       0.673780  0.001630   \n",
              "3950  p.Val709Pro     0.013122     0.048279       0.030700  0.000309   \n",
              "\n",
              "       std_dev  QC_flag functional_consequence  \n",
              "0     0.021163     True                    NaN  \n",
              "1     0.044836     True                    NaN  \n",
              "2     0.012955     True                    NaN  \n",
              "3     0.028860     True                    NaN  \n",
              "4     0.020138     True                    NaN  \n",
              "...        ...      ...                    ...  \n",
              "3946  0.108029     True    functionally_normal  \n",
              "3947  0.018123     True    functionally_normal  \n",
              "3948  0.017522     True    functionally_normal  \n",
              "3949  0.040378     True    functionally_normal  \n",
              "3950  0.017579     True  functionally_abnormal  \n",
              "\n",
              "[3951 rows x 8 columns]"
            ]
          },
          "execution_count": null,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "simple_score = simple_score.merge(\n",
        "    df[[\"hgvs_p\", \"functional_consequence\"]].drop_duplicates(),\n",
        "    left_on=\"variant\",\n",
        "    right_on=\"hgvs_p\",\n",
        "    how=\"left\"\n",
        ").drop(columns=[\"hgvs_p\"])\n",
        "\n",
        "simple_score"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "0c5d6385",
      "metadata": {},
      "outputs": [],
      "source": [
        "simple_score.to_csv(\"20260217_TSC2_lib1_scores_QC.csv\", index=False)\n",
        "print(\"Exported to 20260217_TSC2_lib1_scores_QC.csv\")"
      ]
    }
  ],
  "metadata": {
    "kernelspec": {
      "display_name": "GG_lib",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.13.2"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 5
}
